Source/JavaScriptCore/bytecode/ArrayProfile.h

```diff
@@ -245,27 +247,14 @@ class ArrayProfile {
     static Structure* polymorphicStructure() { return static_cast<Structure*>(reinterpret_cast<void*>(1)); }
     
     StructureID m_lastSeenStructureID;
-    bool m_mayStoreToHole { false }; // This flag may become overloaded to indicate other special cases that were encountered during array access, as it depends on indexing type. Since we currently have basically just one indexing type (two variants of ArrayStorage), this flag for now just means exactly what its name implies.
-    bool m_outOfBounds { false };
-#if USE(LARGE_TYPED_ARRAYS)
-    bool m_mayBeLargeTypedArray { false };
-#endif
-    bool m_mayInterceptIndexedAccesses : 1 { false };
-    bool m_usesOriginalArrayStructures : 1 { true };
-    bool m_didPerformFirstRunPruning : 1 { false };
+    OptionSet<ArrayProfileFlag> m_arrayProfileFlags;
     ArrayModes m_observedArrayModes { 0 };
 };
 static_assert(sizeof(ArrayProfile) == 12);
 
 class UnlinkedArrayProfile {
 public:
-    explicit UnlinkedArrayProfile()
-        : m_usesOriginalArrayStructures(true)
```

Comment:
This is not preserved?

Reply:
To make everything false by default, I flipped this flag to `m_usesNonOriginalArrayStructures`. So this is now fine.